### PR TITLE
OKTA-723081: invalid uri

### LINKED ISSUE #691: Okta Configuration: Invalid URI
### Describe the bug?

Attempting to use SDK for API calls, but cannot get passed `System.UriFormatException: 'Invalid URI: The hostname could not be parsed.'`

Providing code snippet below with secrets redacted.
```c#
var oktaDomain = "https://dev-****.okta.com";
var clientId = ****;

var privateKey = new JsonWebKeyConfiguration
{
    D = ****,
    P = ****,
    Q = ****,
    Dp = ****,
    Dq = ****,
    Qi = ****,
    Kty = "RSA",
    E = "AQAB",
    Kid = ****,
    N = ****
};

var oktaConfiguration = new Configuration
{
    OktaDomain = oktaDomain,
    AuthorizationMode = AuthorizationMode.PrivateKey,
    ClientId = clientId,
    Scopes = new HashSet<string> { "okta.users.read" }, // Add all the scopes you need
    PrivateKey = privateKey,
};            

var userApi = new UserApi(oktaConfiguration);
var users = userApi.ListUsers().ToListAsync().Result;
```

### What is expected to happen?

List of Users is returned

### What is the actual behavior?

The following Exception is thrown back.

`System.UriFormatException: 'Invalid URI: The hostname could not be parsed.'`

### Reproduction Steps?

Generate a private key through a service application.
Fill in the data as demonstrated in the documentation.
Attempt to make an API call with the SDK.

### Additional Information?

_No response_

### .NET Version

7.0.312

### SDK Version

7.0.4

### OS version

- BuildNumber: 19045
- Caption: Microsoft Windows 10 Enterprise
- OSArchitecture: 64-bit
- Version: 10.0.19045

### PR DESCRIPTION
**Pull request by bryanapellanes-okta**
- Add `UseProxy` property to configuration as advice to configuration system in case of unrelated `Proxy` node in configuration.

Fixes #691

### CODE REVIEW

**Comment by bryanapellanes-okta** [comment]
This appears to be sufficient for at least one person experiencing the issue: https://github.com/okta/okta-sdk-dotnet/issues/691#issuecomment-2168494030

**Review by arvindkrishnakumar-okta — APPROVED** [review]

File: src/Okta.Sdk/Client/ApiClient.cs [hunk 1]
```diff
@@ -476,7 +476,7 @@ internal RestClient GetConfiguredClient(IReadableConfiguration configuration)
             {
                 clientOptions.Proxy = _proxy;
             }
-            else if (configuration.Proxy != null)
+            else if (configuration.Proxy != null && configuration.UseProxy == true)
```
Comment (laura-rodriguez) [inline src/Okta.Sdk/Client/ApiClient.cs:479]: It seems UseProxy is nullable, should we add a null check here?

**Review by laura-rodriguez — COMMENTED** [review]
We should update the configuration section in the readme file indicating how to use proxy in 8.0.1+ version

File: src/Okta.Sdk/Client/ApiClient.cs (hunk 1 above)
Comment (bryanapellanes-okta) [reply src/Okta.Sdk/Client/ApiClient.cs:479]: If UseProxy is null in the comparison then it will be `null == true` which evaluates to false.

Comment (laura-rodriguez) [reply src/Okta.Sdk/Client/ApiClient.cs:479]: Right!

File: README.md [hunk 2]
```diff
@@ -510,6 +510,15 @@ okta:
     rateLimit:
       maxRetries: 4
 ```
+ 
+ Beginning in version 8.0.1, If you have need for a `proxy` node in your configuration unrelated to a web proxy for the Okta API client, set `useProxy` to `false`:
```
Comment (laura-rodriguez) [inline README.md:514]:
```suggestion
 Beginning in version 8.0.1, If you have need for a `proxy` node in your configuration unrelated to a web proxy for the Okta API client or you want to disable the proxy without removing it from your configuration, set `useProxy` to `false`:
```

**Review by laura-rodriguez — DISMISSED** [review]

**Pull request merged**